reactor-core/src/main/java/reactor/core/publisher/FluxHandle.java

```diff
@@ -106,10 +113,15 @@ public void onNext(T t) {
 				actual.onNext(v);
 			}
 			if(stop){
-				s.cancel();
 				if(error != null){
-					onError(Operators.onOperatorError(null, error, t,
-							actual.currentContext()));
+					Throwable e_ = Operators.onNextError(t, error, actual.currentContext(), s);
+					if (e_ != null) {
+						onError(e_);
+					}
+					else {
+						reset();
+						s.request(1L);
+					}
 					return;
 				}
```

Comment:
I think it would need the `s.cancel` after this block (from [original line 109](https://github.com/reactor/reactor-core/pull/957/files#diff-038647da5050f6dfe3d2b29fd8e6aa43L109))

Reply:
Cancellation happens in `Operators.onNextError` if the error is handled, so operator itself can't cancel prior to invoking strategy.

Reply:
the old code would always cancel `s`, whereas with your modification it only cancels when there is an error without continue mode, not when there is completion.